script/common.bash

```diff
@@ -41,6 +41,19 @@ get_go_version() {
 setup_go_root() {
   local version=${1}
 
+  # Use the current Go installation if the given Go version is already
+  # installed and configured.
+  if command -v go ; then
+    debug "Found Go. Checking version..."
+    set +e
+    FOUND_GO_VERSION=$(go version|awk '{print $3}'|sed s/go//)
+    set -e
+    if [ "$FOUND_GO_VERSION" == "$version" ] ; then
+      debug "Versions match. No need to install Go. Exiting."
+      exit 0
+    fi
+  fi
+
```

Comment:
This is something we already do in the `Beats` project:

- https://github.com/elastic/beats/blob/f14f67eb0acfa845c27029a42bcba03eff4c6f36/.ci/scripts/install-go.sh#L12-L23